From 7c9836880252d7347502b7ca20d4c72e22feba41 Mon Sep 17 00:00:00 2001
From: Till Kamppeter <till.kamppeter@gmail.com>
Date: Sat, 15 Mar 2025 21:23:57 +0100
Subject: [PATCH 01/33] cfFilterTextToPDF(): Let all Arabic characters be
 rendered right-to-left

Fixes #84

In the Arabic languages text is written right-to-left, not
left-to-right as Latin characters are written.

This fix makes all Arabic characters correctly rendered right-to-left.
---
 charset/pdf.utf-8.heavy  | 1 +
 charset/pdf.utf-8.simple | 1 +
 2 files changed, 2 insertions(+)

diff --git a/charset/pdf.utf-8.heavy b/charset/pdf.utf-8.heavy
index a610af01..6273d6c5 100644
--- a/charset/pdf.utf-8.heavy
+++ b/charset/pdf.utf-8.heavy
@@ -33,6 +33,7 @@ charset utf8
 0300 03FF ltor single DejaVuSansMono
 0400 04FF ltor single DejaVuSansMono DejaVuSansMono:bold DejaVuSansMono:oblique DejaVuSansMono:bold:oblique
 0500 05FF rtol single FreeMono
+0600 06FF rtol single FreeMono
 1E00 1EFF ltor single CourierNew CourierNew:bold CourierNew:italic CourierNew:bold:italic
 2000 21FF ltor single DejaVuSansMono DejaVuSansMono:bold DejaVuSansMono:oblique DejaVuSansMono:bold:oblique
 2200 23FF ltor single Symbol
diff --git a/charset/pdf.utf-8.simple b/charset/pdf.utf-8.simple
index f7d3adf7..42d1af0a 100644
--- a/charset/pdf.utf-8.simple
+++ b/charset/pdf.utf-8.simple
@@ -30,4 +30,5 @@ charset utf8
 
 0000 04FF ltor single monospace monospace:bold monospace:oblique monospace:bold:oblique
 0500 05FF rtol single monospace
+0600 06FF rtol single monospace
 3000 9FFF ltor double ARPLUmingCN
-- 
2.53.0

